### PR TITLE
[noticket] Add possibility for inidividual asset paths

### PR DESCRIPTION
**Pull request by lx-wnk**
### 1. Why is this change necessary?
To set individual paths for bundles

### 2. What does this change do, exactly?
Allows the developer to set custom paths for administration/storefront and snippet paths

### 3. Describe each step to reproduce the issue or behaviour.
Try to

### 4. Please link to the relevant issues (if any).
///

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: changelog/_unreleased/2020-09-14-Adjust-index-cleanup.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+ ---
+ title:              Add customizable paths in plugin
+ issue:              
+ author:             Alexander Wink
+ author_email:       a.wink@kellerkinder.de
+ author_github:      @jinnoflife
+ ---
+ # Core
+ * Add customizable paths for storefront, admin, view and snippets
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-14-Adjust-index-cleanup.md]:
```suggestion
 * Added customizable paths for storefront, admin, view and snippets 
```

File: changelog/_unreleased/2020-09-14-Adjust-index-cleanup.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+ ---
+ title:              Add customizable paths in plugin
+ issue:
```
Comment (mitelg) [inline changelog/_unreleased/2020-09-14-Adjust-index-cleanup.md]: remove empty fields

File: src/Core/Framework/Bundle.php [hunk 3]
```diff
@@ -100,20 +105,27 @@ public function configureRouteOverwrites(RouteCollectionBuilder $routes, string
...
-     */
-    protected function getActionEvents(): array
+    public function getViewPaths(): array
     {
-        return [];
+        return [self::DEFAULT_PATH_VIEWS];
     }
 
-    /**
-     * Returns a list of all action event class references of this bundle. The events will be registered inside the `\Shopware\Core\Framework\Event\BusinessEventRegistry`.
```
Comment (mitelg) [inline src/Core/Framework/Bundle.php]: why did you remove the comments?

File: src/Core/Framework/Bundle.php [hunk 4]
```diff
@@ -100,20 +105,27 @@ public function configureRouteOverwrites(RouteCollectionBuilder $routes, string
         }
     }
 
-    /**
-     * @feature-deprecated tag:v6.4.0 - Implement `getActionEventClasses` instead
```
Comment (mitelg) [inline src/Core/Framework/Bundle.php:104]: why did you remove the deprecation?

File: src/Core/Framework/Bundle.php [hunk 5]
```diff
@@ -159,7 +171,6 @@ private function registerEvents(ContainerBuilder $container): void
     {
         $definition = $container->getDefinition(BusinessEventRegistry::class);
         $definition->addMethodCall('addMultiple', [$this->getActionEvents()]);
-        $definition->addMethodCall('addClasses', [$this->getActionEventClasses()]);
```
Comment (mitelg) [inline src/Core/Framework/Bundle.php]: why did you remove that?

Comment (lx-wnk) [reply src/Core/Framework/Bundle.php]: I'm not sure why this was removed. I just added it again

File: src/Core/Framework/Bundle.php (hunk 4 above)
Comment (lx-wnk) [reply src/Core/Framework/Bundle.php:104]: I'm not sure why this was removed. I just added it again

File: src/Core/Framework/Bundle.php (hunk 3 above)
Comment (lx-wnk) [reply src/Core/Framework/Bundle.php]: I'm not sure why those were removed. I just them again

File: changelog/_unreleased/2020-09-14-Adjust-index-cleanup.md (hunk 2 above)
Comment (lx-wnk) [reply changelog/_unreleased/2020-09-14-Adjust-index-cleanup.md]: done

File: changelog/_unreleased/2020-09-14-Adjust-index-cleanup.md (hunk 1 above)
Comment (lx-wnk) [reply changelog/_unreleased/2020-09-14-Adjust-index-cleanup.md]: done

**Comment by mitelg** [comment]
hey @jinnoflife 

I'm sorry to say that now, but after talking to my colleagues, we could not merge your PR :neutral_face: 
We did have such functionality in the past and removed it again, because it brings some complexity with it. Also the chance, that this is configured wrong by plugin developers is quite high, which will lead to unexpected behaviour, that is hard to debug. 

thank you anyway for your contribution :blue_heart: keep up the good work :blush: :+1:
